TelCo.ColorCoder/ColorMap.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Diagnostics;
+using System.Drawing;
+
+namespace TelCo.ColorCoder
+{
+        /// <summary>
+        /// data type defined to hold the two colors of clor pair
+        /// </summary>
+        internal class ColorPair
+        {
+            internal Color majorColor;
+            internal Color minorColor;
+            public override string ToString()
+            {
+                return string.Format("MajorColor:{0}, MinorColor:{1}", majorColor.Name, minorColor.Name);
+            }
+        }
+        
+        class ColorMap{
```

Comment:
A general guideline is have the name of file and class name to be same. Have single class in a single file.
Unless you wish to have have multiple small classes within a file. In which class the file name should indicate the same. Here you should probably call it ColorCodeDataStructures or something.

Better is to have separate file for each class.